src/ucs/sys/netlink.c

```diff
@@ -300,13 +283,65 @@ int ucs_netlink_route_exists(int if_index, const struct sockaddr *sa_remote,
                                  sizeof(rtm), ucs_netlink_parse_rt_entry_cb,
                                  NULL);
     }
+}
+
+static void ucs_netlink_lookup_route(ucs_netlink_route_info_t *info)
+{
+    ucs_netlink_rt_rules_t *iface_rules;
+    khiter_t iter;
+
+    ucs_netlink_init_routing_table_cache();
+
+    iter = kh_get(ucs_netlink_rt_cache, &ucs_netlink_routing_table_cache,
+                  info->if_index);
+    if (iter == kh_end(&ucs_netlink_routing_table_cache)) {
+        return;
+    }
+
+    iface_rules = &kh_val(&ucs_netlink_routing_table_cache, iter);
+    info->netmask_len = ucs_netlink_lookup_in_iface_rules(info->sa_remote,
+                                                                iface_rules);
+}
+
+static int ucs_netlink_max_netmask_len(const struct sockaddr *sa_remote)
+{
+    int max_netmask_len = -1;
+    ucs_netlink_rt_rules_t iface_rules;
+
+    kh_foreach_value(&ucs_netlink_routing_table_cache, iface_rules, {
+        int curr_netmask_len = ucs_netlink_lookup_in_iface_rules(sa_remote,
+                                                                 &iface_rules);
+        if (curr_netmask_len > max_netmask_len) {
+            max_netmask_len = curr_netmask_len;
+        }
+    });
 
-    info.if_index         = if_index;
-    info.sa_remote        = sa_remote;
-    info.found            = 0;
-    info.allow_default_gw = allow_default_gw;
+    return max_netmask_len;
+}
+
+int ucs_netlink_route_exists(int if_index, const struct sockaddr *sa_remote,
+                             int *netmask_len_p)
+{
+    ucs_netlink_route_info_t info = {.if_index    = if_index,
+                                     .sa_remote   = sa_remote,
+                                     .netmask_len = -1};
```

Comment:
Minor: formatting
```suggestion
    ucs_netlink_route_info_t info = {
        .if_index    = if_index,
        .sa_remote   = sa_remote,
         .netmask_len = -1
  };
```

It seems that there are multiple clang-format issues
check `git-clang-format --diff HEAD~13`

Other than that, it's approved on my side